cmd/influxd/inspect/report_db/report_db.go

```diff
@@ -0,0 +1,190 @@
+package report_db
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"text/tabwriter"
+
+	"github.com/influxdata/influxdb/v2/cmd/influxd/inspect/report_db/aggregators"
+	"github.com/influxdata/influxdb/v2/kit/cli"
+	"github.com/influxdata/influxdb/v2/models"
+	"github.com/influxdata/influxdb/v2/pkg/reporthelper"
+	"github.com/influxdata/influxdb/v2/tsdb/engine/tsm1"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+	"golang.org/x/sync/errgroup"
+)
+
+// ReportDB represents the program execution for "influxd report-db".
+type ReportDB struct {
+	// Standard input/output, overridden for testing.
+	Stderr io.Writer
+	Stdout io.Writer
+
+	dbPath     string
+	shardPaths map[uint64]string
+	exact      bool
+	detailed   bool
+	// How many goroutines to dedicate to calculating cardinality.
+	concurrency int
+	// t, d, r, m for Total, Database, Retention Policy, Measurement
+	rollup string
+}
+
+func NewReportDBCommand(v *viper.Viper) (*cobra.Command, error) {
+	flags := &ReportDB{
+		Stderr: os.Stderr,
+		Stdout: os.Stdout,
+	}
+
+	cmd := &cobra.Command{
+		Use:   "report-db",
+		Short: "Estimates cloud 2 cardinality for a database",
+		Args:  cobra.NoArgs,
+		RunE: func(cmd *cobra.Command, _ []string) error {
+			return reportDBRunE(cmd, flags)
+		},
+	}
+	opts := []cli.Opt{
+		{
+			DestP:    &flags.dbPath,
+			Flag:     "db-path",
+			Desc:     "path to database",
+			Required: true,
+		},
+		{
+			DestP:   &flags.concurrency,
+			Flag:    "c",
+			Desc:    "set worker concurrency, defaults to one",
+			Default: 1,
+		},
+		{
+			DestP:   &flags.detailed,
+			Flag:    "detailed",
+			Desc:    "include counts for fields, tags",
+			Default: false,
+		},
+		{
+			DestP:   &flags.exact,
+			Flag:    "exact",
+			Desc:    "report exact counts",
+			Default: false,
+		},
+		{
+			DestP:   &flags.rollup,
```

Comment:
Does this work for 2.0 with buckets and stuff?  Should the arguments to rollup level be different?

Reply:
Is the structure org/bucket/measurement?

Reply:
Or are orgs not in the directory structure?  

https://docs.influxdata.com/influxdb/v2.5/reference/internals/file-system-layout/?t=Linux#tsm-directories-and-files-layout

Reply:
It does work for 2.0, though I did need to rename a few things to have it make sense for 2.0 (pushed a new commit for that).

1.x: `db_name/rp_name/shard_num/tsm_file`
2.x: `bucket_name/rp_name/shard_num/tsm_file`

<img width="732" alt="Screenshot 2022-11-18 at 08 39 54" src="https://user-images.githubusercontent.com/26460069/202724803-7f62d7ae-6cd5-43b5-8c56-c722efa53775.png">